client/utils/useHistoricalLockupToasts.js

```diff
@@ -40,17 +39,13 @@ const useHistoricalLockupToasts = () => {
     const ogvLockedUp = event.args[2];
     const lockUpEnd = parseInt(event.args[3]);
     const blockTime =
-      (await web3Provider?.getBlock(event.blockNumber)).timestamp ||
+      (await rpcProvider?.getBlock(event.blockNumber)).timestamp ||
```

Comment:
If rpcProvider is `null` or `undefined`, won't this try to read `undefined.timestamp`? Would probably suggest changing it to `(await rpcProvider?.getBlock(event.blockNumber))?.timestamp` (i.e. add the `?` operator outside the brackets as well)

Reply:
This was a bit legacy code but I can adjust this